examples/bzlmod/MODULE.bazel

```diff
@@ -1,36 +1,143 @@
+local_path_override(
+    module_name = "rules_python_gazelle_plugin",
+    path = "../../gazelle",
+)
+
+# The following stanza defines the dependency for gazelle
+# See here https://github.com/bazelbuild/bazel-gazelle/releases/ for the
+# latest version.
+bazel_dep(name = "gazelle", version = "0.30.0", repo_name = "bazel_gazelle")
+
+# The following stanze returns a proxy object representing a module extension;
+# its methods can be invoked to create module extension tags.
 python = use_extension("@rules_python//python:extensions.bzl", "python")
+
+# This name is passed into python.toolchain and it's use_repo statement.
+# We also use the same name for python.host_python_interpreter.
+PYTHON_NAME = "python3"
+
+# This is the name that is used for the host interpreter
+PYTHON_INTERPRETER = PYTHON_NAME + "_host_interpreter"
+
+# We next initialize the python toolchain using the extension.
+# You can set different Python versions in this block.
 python.toolchain(
-    name = "python3_9",
+    # This name is used in the various use_repo statements
+    # below, and in the local extension that is in this
+    # example.
+    name = PYTHON_NAME,
     configure_coverage_tool = True,
     python_version = "3.9",
 )
-use_repo(python, "python3_9")
-use_repo(python, "python3_9_toolchains")
 
+# Import the python repositories generated by the given module extension
+# into the scope of the current module.
+# All of the python3 repositories use the PYTHON_NAME as there prefix.  They
+# are not catenated for ease of reading.
+use_repo(python, PYTHON_NAME)
+use_repo(python, "python3_toolchains")
+
+# In bazel 6.2 you will be able to load these dynamically.
+# We need each of these repos load so that for setting the pyton_interpreter_target.
+# You only need the Python repo for the operating systems that you are supporting.
+use_repo(python, "python3_x86_64-unknown-linux-gnu")
+use_repo(python, "python3_aarch64-apple-darwin")
+use_repo(python, "python3_aarch64-unknown-linux-gnu")
+use_repo(python, "python3_x86_64-apple-darwin")
+use_repo(python, "python3_x86_64-pc-windows-msvc")
+
+# This extension is used to look up the host system interpreter.
+# A label that is created by this extension is used for the
+# python_interpreter_target.
+python.host_python_interpreter(
+    # The name sets the repo name that is used below
+    # We then have a custom interpreter name that we use
+    # in the pip arguments below.
+    name = PYTHON_INTERPRETER,
+    # You only need the interpreters that you are supporting. In your project.
+    # These labels are referencing the above repos, and they all have a :python
+    # exported.
+    interpreters = [
+        "@python3_x86_64-unknown-linux-gnu//:python",
+        "@python3_aarch64-apple-darwin//:python",
+        "@python3_aarch64-unknown-linux-gnu//:python",
+        "@python3_x86_64-apple-darwin//:python",
+        "@python3_x86_64-pc-windows-msvc//:python",
+    ],
```

Comment:
I think we can get rid of having to pass these in. I was talking to a Xudong about this problem, and he said it sounded like a bug and asked for a simple repro. [When I tried][1]...I couldn't repro it and didn't need to pass these labels in for them to resolve. I'm going to take another stab at figuring out why.

[1]: https://github.com/bazelbuild/rules_python/compare/main...rickeylev:rules_python:repo-resolve-bug?expand=1

Reply:
How do we know what to check if we don't pass them in?